database/models.go

```diff
@@ -22,47 +22,128 @@ import (
 	"github.com/coreos/clair/pkg/pagination"
 )
 
-// Processors are extentions to scan a layer's content.
-type Processors struct {
-	Listers   []string
-	Detectors []string
-}
-
 // Ancestry is a manifest that keeps all layers in an image in order.
 type Ancestry struct {
+	// Name is a non-empty string to uniquely identify the Ancestry.
 	Name string
-	// ProcessedBy contains the processors that are used when computing the
+	// By contains the processors that are used when computing the
 	// content of this ancestry.
-	ProcessedBy Processors
+	By []Detector
 	// Layers should be ordered and i_th layer is the parent of i+1_th layer in
 	// the slice.
 	Layers []AncestryLayer
 }
 
+// Valid checks if the ancestry is compliant to spec.
+func (a *Ancestry) Valid() bool {
+	if a == nil {
+		return false
+	}
+
+	if a.Name == "" {
+		return false
+	}
+
+	for _, d := range a.By {
+		if !d.Valid() {
+			return false
+		}
+	}
+
+	for _, l := range a.Layers {
+		if !l.Valid() {
+			return false
+		}
+	}
+
+	return true
+}
+
 // AncestryLayer is a layer with all detected namespaced features.
 type AncestryLayer struct {
-	LayerMetadata
-
-	// DetectedFeatures are the features introduced by this layer when it was
+	// Hash is the name of the ancestry's layer.
+	Hash string
```

Comment:
TODO: We need to address that Hash MUST be a content addressable hash for the layer tar ball.

Reply:
TODO: Add specification: hash must be non-empty

Reply:
This comment is wrong, Hash is on the Ancestry Layer NOT the Ancestry.